proxy/pkg/middleware/accesslog.go

```diff
@@ -18,7 +20,7 @@ func AccessLog(logger log.Logger) func(http.Handler) http.Handler {
 
 			logger.Info().
 				Str("proto", r.Proto).
-				Str("request", r.Header.Get("X-Request-ID")).
+				Str("request", ExtractRequestID(r.Context())).
```

Comment:
I just saw that chi middleware also provides this method. You could use:
```suggestion
				Str("request", chimiddleware.GetReqID(r.Context())).
```